src/utils.cpp

```diff
@@ -91,6 +91,21 @@ std::vector<std::string> expand_wildcard_paths(
   return expanded_paths;
 }
 
+uint32_t round_up_to_next_power_of_two(uint32_t n)
+{
+  // http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+  if (n == 0) {
+    return 0;
+  }
+  n--;
+  n |= n >> 1;
+  n |= n >> 2;
+  n |= n >> 4;
+  n |= n >> 8;
+  n |= n >> 16;
+  return n + 1;
+}
+
```

Comment:
Might be good to add some unit tests for posterity. At least it'll be documentation

Reply:
Done, even found I had to deal with edge cases where n > 2^31 :)